### PR TITLE
[DROOLS-5783] Make droolsjbpm-integration repo independent from appformer

### PR DESCRIPTION
**Pull request by danielezonca**
https://issues.redhat.com/browse/DROOLS-5783

Description:
The goal of this PR is to cleanup `droolsjbpm-integration` repo from unnecessary relation with other repos (in this case `appformer`). It is a sort of followup of this PR https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1505 that tries to do the same but it has been partially reverted because of some remaining usages

Details of this part of the PR:
- moving `kie-identity-session-provider` from `droolsjbpm-integration` repo to here because it is used by some -wb repos but it uses kie-internal so this is the first repo in common we can use

NOTE: I preserved same group and artifact of `kie-identity-session-provider` to preserve downstream usages. If we don't like to have a different naming I can change it + all downstreaming repos

@mareknovotny @Ginxo @lucamolteni @mariofusco @ederign 

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1517
* https://github.com/kiegroup/kie-soup/pull/163
* https://github.com/kiegroup/appformer/pull/1073
* https://github.com/kiegroup/droolsjbpm-integration/pull/2306
* https://github.com/kiegroup/kie-wb-common/pull/3472

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Review by lucamolteni — APPROVED** [review]

**Review by adrielparedes — APPROVED** [review]

**Comment by mareknovotny** [comment]
@danielezonca there are few sonarcloud failures we could fix and have better source code. I know you are not the author of the code, but I think this is simple to fix now. WDYT?

File: kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeIdentityProvider.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+*/
+
+package org.kie.provider.impl;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import javax.enterprise.context.RequestScoped;
```
Comment (mareknovotny) [inline kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeIdentityProvider.java:23]: not used anywhere?
```suggestion

```

File: kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeIdentityProvider.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    private String getIdentityFromRequest() { 
+        if (!request.isUnsatisfied() && request.get().getUserPrincipal() != null) {
+            return request.get().getUserPrincipal().getName();
+        }
+        return UNKNOWN_USER_IDENTITY;
+    }
+    
+    @Override
+    public List<String> getRoles() {
+        List<String> roles = new ArrayList<String>();
```
Comment (mareknovotny) [inline kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeIdentityProvider.java:74]: just minor thing, but sonar will be happy?
```suggestion
        List<String> roles = new ArrayList<>();
```

File: kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeSessionInfoProvider.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import org.uberfire.rpc.SessionInfo;
+import org.uberfire.rpc.impl.SessionInfoImpl;
+
+@SessionScoped
+public class SafeSessionInfoProvider implements SessionInfoProvider, Serializable  {
+
+    /** Generated serial version UID */
+    private static final long serialVersionUID = 8510219062936244657L;
+
+    private static final Logger logger = LoggerFactory.getLogger( SafeIdentityProvider.class );
```
Comment (mareknovotny) [inline kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeSessionInfoProvider.java:43]: A copy&paste error from past I think 

```suggestion
    private static final Logger logger = LoggerFactory.getLogger( SafeSessionInfoProvider.class );
```

File: kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeSessionInfoProvider.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            return delegate.get().getIdentity(); 
+        } catch ( Exception e ) {
+            logger.debug("SessionInfo bean was available but could not return identity: " + e.getMessage(), e );
+            return getUserFromIdentityProvider();
+        }
+    }
+    
+    private User getUserFromIdentityProvider() { 
+        List<String> roleStrList = identityProvider.getRoles();
+        List<Role> roles = new ArrayList<Role>(roleStrList.size());
```
Comment (mareknovotny) [inline kie-identity-session-provider/src/main/java/org/kie/provider/impl/SafeSessionInfoProvider.java:82]: shorter version

```suggestion
        List<Role> roles = new ArrayList<>(roleStrList.size());
```

**Review by mareknovotny — CHANGES_REQUESTED** [review]
Suggested some problems from original moved source code, You can simply accept it to reduce sonarcloud failures.

**Comment by danielezonca** [comment]
@mareknovotny 
Thanks for the suggestions, accepted :+1:

**Review by mariofusco — APPROVED** [review]

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**